graphics/managed_surface.h

```diff
@@ -61,18 +61,43 @@ class ManagedSurface {
 	 * the offset from the parent's top-left corner this sub-surface starts at
 	 */
 	Common::Point _offsetFromOwner;
-public:
+
+	/**
+	 * Automatic transparency color. When set, doesn't require transparency to be
+	 * explicitly passed, and blit methods pass on to transBlit
+	 */
+	uint _transparentColor;
+	bool _transparentColorSet;
+
+	/**
+	 * Local palette for 8-bit images
+	 */
+	uint32 _palette[256];
+	bool _paletteSet;
+protected:
```

Comment:
This breaks at least Director engine. Why is this declared protected now?

Reply:
Whoops, sorry. I'd gotten so used to just compiling with only the engine enabled, I forgot to ensure that my changes to the core didn't break anything else. I'll do an interactive rebase this evening and fix it in the history.

Reply:
Regarding lua, any help would be welcome. I have a branch ultima6_lua on my github repository that has a single commit at the head where I was experimenting with extensions to the core Lua to re-enable proper file loading again. That could be a good starting point for any experiments.

Reply:
The branch is a bit out of date though, so I'll also rebase it against the current ultima branch tonight.